### PR TITLE
Fix typo in MessageFactory

### PR DESCRIPTION
**Pull request by MeladKamari**
fix typo in MessageFactory

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/orleans/pull/7901)

### CODE REVIEW

**Comment by ReubenBond** [comment]
You can't just call `Task.Delay(x)` without awaiting it: it will do nothing useful in that case.

File: src/Orleans.CodeGenerator.MSBuild/Program.cs [hunk 1]
```diff
@@ -52,7 +53,7 @@ public static int Main(string[] args)
                                     Console.WriteLine("Waiting for debugger to attach.");
                                 }
 
-                                Thread.Sleep(100);
+                                Task.Delay(100);
```
Comment (ReubenBond) [inline src/Orleans.CodeGenerator.MSBuild/Program.cs:56]: `Task.Delay` must be awaited for it to have the desired effect.

**Comment by ReubenBond** [comment]
By the way, there is no need to change `Thead.Sleep` to `Task.Delay` in these cases. In some cases (eg, `Watchdog`) it is *better* to use `Thread.Sleep`, since it allows the watchdog to continue despite issues such as Thread Pool starvation or timer queue lock contention. Both of those are less of an issue in newer .NET versions than they were but can still bite - and the watchdog should keep ticking.

File: src/Orleans.CodeGenerator.MSBuild/Program.cs (hunk 1 above)
Comment (MeladKamari) [reply src/Orleans.CodeGenerator.MSBuild/Program.cs]: Thanks reuben , i revert commit

**Review by ReubenBond — APPROVED** [review]

**Comment by ReubenBond** [comment]
Thank you, @MeladKamari!

**Pull request merged**